src/dune/string_with_vars.mli

```diff
@@ -114,14 +114,24 @@ val fold_vars : t -> init:'a -> f:(Var.t -> 'a -> 'a) -> 'a
 
 type 'a expander = Var.t -> Dune_lang.Syntax.Version.t -> 'a
 
-val expand :
-  t -> mode:'a Mode.t -> dir:Path.t -> f:Value.t list option expander -> 'a
-
-val partial_expand :
-     t
-  -> mode:'a Mode.t
-  -> dir:Path.t
-  -> f:Value.t list option expander
-  -> 'a Partial.t
+module type S = sig
```

Comment:
Can you add some comments here? What do these two functions do, and why do we need to parameterise over the applicative?

Reply:
Sure, I'll add some comments. The basic idea is that we'd like to accumulate a `Build.t` while expanding.

Reply:
Thanks! The (almost complete) lack of comments in Dune makes it really hard for me to understand various key mechanisms.